tables/AlternateMans/test/tUvwFile.cc

```diff
@@ -0,0 +1,187 @@
+  {1, 4},
+  {3, 3},
+  {3, 4},
+  {4, 4}
+};
+}
+
+BOOST_AUTO_TEST_SUITE(uvw_file)
+
+BOOST_AUTO_TEST_CASE(create_empty) {
+  UvwFile file_a = UvwFile::CreateNew(kFilename);
+  BOOST_CHECK_EQUAL(file_a.NRows(), 0);
+  BOOST_CHECK_EQUAL(file_a.Filename(), kFilename);
+  file_a.Close();
+
+  UvwFile file_b = UvwFile::CreateNew(kFilename);
+  file_b = std::move(file_a);
+  BOOST_CHECK_EQUAL(file_b.NRows(), 0);
+  BOOST_CHECK_EQUAL(file_b.Filename(), "");
+  file_b.Close();
+
+  unlink(kFilename.c_str());
+}
+
+BOOST_AUTO_TEST_CASE(open_empty) {
+  UvwFile file = UvwFile::CreateNew(kFilename);
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), kFilename);
+  file.Close();
+
+  file = UvwFile::OpenExisting(kFilename);
+  BOOST_CHECK_EQUAL(file.NRows(), 0);
+  BOOST_CHECK_EQUAL(file.Filename(), kFilename);
+  file.Close();
+
+  unlink(kFilename.c_str());
+}
+
+void WriteTwoTimesteps(UvwFile& file, const std::array<std::size_t, 2>* baselines, bool with_autos) {
+  uint64_t row = 0;
+  for(uint64_t i=0; i!=kNBaselines; ++i) {
+    if(with_autos || baselines[i][0] != baselines[i][1]) {
+      const std::array<double, 3>& uvw = kUvwPerBaseline[i];
+      file.WriteUvw(row, baselines[i][0], baselines[i][1], uvw.data());
+      BOOST_CHECK_EQUAL(file.NRows(), row+1);
+      ++row;
+    }
+  }
+  for(uint64_t i=0; i!=kNBaselines; ++i) {
+    if(with_autos || baselines[i][0] != baselines[i][1]) {
+      std::array<double, 3> uvw = kUvwPerBaseline[i];
+      uvw[0] *= 2.0;
+      uvw[1] *= 3.0;
+      uvw[2] *= 4.0;
+      file.WriteUvw(row, baselines[i][0], baselines[i][1], uvw.data());
+      BOOST_CHECK_EQUAL(file.NRows(), row+1);
+      ++row;
+    }
+  }
+}
+
+void ReadTwoTimesteps(UvwFile& file, const std::array<std::size_t, 2>* baselines, bool with_autos) {
+  // Read the second timestep first to see if seeking works ok.
+  uint64_t row = with_autos ? kNBaselines : 3;
+  for(uint64_t i=0; i!=kNBaselines; ++i) {
+    if(with_autos || baselines[i][0] != baselines[i][1]) {
+      std::array<double, 3> uvw;
+      file.ReadUvw(row, baselines[i][0], baselines[i][1], uvw.data());
+      std::array<double, 3> expected_uvw = kUvwPerBaseline[i];
+      expected_uvw[0] *= 2.0;
+      expected_uvw[1] *= 3.0;
+      expected_uvw[2] *= 4.0;
+      for(size_t z=0; z!=3; ++z)
+        BOOST_CHECK_CLOSE_FRACTION(expected_uvw[z], uvw[z], 1e-6);
+      ++row;
+    }
+  }
+  row = 0;
+  for(uint64_t i=0; i!=kNBaselines; ++i) {
+    if(with_autos || baselines[i][0] != baselines[i][1]) {
+      std::array<double, 3> uvw;
+      file.ReadUvw(row, baselines[i][0], baselines[i][1], uvw.data());
+      for(size_t z=0; z!=3; ++z)
+        BOOST_CHECK_CLOSE_FRACTION(kUvwPerBaseline[i][z], uvw[z], 1e-6);
+      ++row;
+    }
+  }
+}
+
+BOOST_AUTO_TEST_CASE(write_and_direct_read) {
+  UvwFile file = UvwFile::CreateNew(kFilename);
+  WriteTwoTimesteps(file, kBaselines, true);
+  ReadTwoTimesteps(file, kBaselines, true);
+  unlink(kFilename.c_str());
```

Comment:
Shouldn't you `file.Close();` before unlinking it? Also in the other tests?

Reply:
Done